src/lib.rs

```diff
@@ -891,6 +922,8 @@ where
 {
     let options: Options = width_or_options.into();
 
+    let line_ending_pat = options.line_ending.as_str();
```

Comment:
I would call this `line_ending_str` like you did elsewhere (it's not really a pattern).